lib/posix/c_lib_ext/fnmatch.c

```diff
@@ -58,8 +59,112 @@ static inline int foldcase(int ch, int flags)
 	return ch;
 }
 
+/**
+ * @brief Function to check for FNM_PERIOD condition
+ *
+ * @param string string whose first character is checked if it is a period
+ * @param flags passed flags to check for FNM_PERIOD
+ * @return int
+ * @retval 1 if first character is period and FNM_PERIOD flag is set
+ * @retval 0 otherwise
+ */
+static int check_fnm_period(const char *string, const int flags)
+{
+	return *string == '.' && (flags & FNM_PERIOD);
+}
+
+/**
+ * @brief Function to check for FNM_PATHNAME condition
+ *
+ * @param letter letter to be checked if it is a '/'
+ * @param flags flags passed to check for FNM_PATHNAME
+ * @return int
+ * @retval 1 if letter is '/' and FNM_PATHNAME flag is set
+ * @retval 0 otherwise
+ */
+static int check_for_pathname(const char letter, const int flags)
+{
+	return letter == '/' && (flags & FNM_PATHNAME);
+}
+
 #define FOLDCASE(ch, flags) foldcase((unsigned char)(ch), (flags))
 
+/**
+ * @brief Custom is_lower function implementation due to problems with
+ * standard library implementation on some platforms
+ *
+ * @param a of type int, to satisfy the islower function signature
+ * @return int
+ * @retval 1 if a is a lowercase letter
+ * @retval 0 otherwise
+ */
+static int is_lower(int a)
```

Comment:
TBH, I think we don't really need this.

See https://github.com/zephyrproject-rtos/zephyr/pull/97400/files#r2443389349 for details.